src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -146,7 +158,7 @@ private void ReloadScript()
                 (_functionSignature == null ||
                 (_functionSignature.HasLocalTypeReference || !_functionSignature.Equals(signature))))
             {
-                Host.RestartEvent.Set();
+                Host.Restart();
```

Comment:
Why do we restart in this case for a successful compilation? Probably the code comment could be better - it's confusing :)